docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,208 @@
+   bytes)
+3. No children columns
+
+As another example, a `StringColumn` backing the Series `['do', 'you', 'have', 'any', 'cheese?']` is composed of:
+
+1. No data buffer
+2. No mask buffer as there are no nulls in the Series
+3. Two children columns:
+
+   - A column of UTF-8 characters
+     `['d', 'o', 'y', 'o', 'u', 'h' ..., '?']`
+   - A column of "offsets" to the characters column (in this case,
+     `[0, 2, 5, 9, 12, 19]`)
+
+
+### Data types
+
+cuDF uses [dtypes](https://numpy.org/doc/stable/reference/arrays.dtypes.html) to represent different types of data.
+Since efficient GPU algorithms require preexisting knowledge of data layouts,
+cuDF does not support the arbitrary `object` dtype, but instead defines a few custom types for common use-cases:
+- `ListDtype`: Lists where each element in every list in a Column is of the same type
+- `StructDtype`: Dicts where a given key always maps to values of the same type
+- `CategoricalDtype`: Analogous to the pandas categorical dtype except that the categories are stored in device memory
+- `DecimalDtype`: Fixed-point numbers
+- `IntervalDtype`: Intervals
+
+Note that there is a many-to-one mapping between data types and `Column` classes.
+For instance, all numerical types (floats and ints of different widths) are all managed using `NumericalColumn`.
+
+
+### Buffer
+
+Although a `Column` represents an Arrow-compliant data structure, it does not directly handle memory mangaement.
+That job is delegated to the `Buffer` class, which represents a device memory allocation that it _may or may not_ own.
+A `Buffer` constructed from a preexisting device memory allocation (such as a CuPy array) will view that memory.
+Conversely, a `Buffer` constructed from a host object will allocate new device memory and copy in the data.
+cuDF uses the [RMM](https://github.com/rapidsai/rmm) library for allocating device memory.
+You can read more about device memory allocation with RMM [here](https://github.com/rapidsai/rmm#devicebuffers).
+
+```{note}
+cuDF needs to interoperate with a wide range of Python libraries, many of which allocate device memory.
+The ownership behavior described above is designed to minimize new memory allocations for externally-owned memory.
+cuDF's use of `libcudf`, however, represents a slight break from the paradigm employed above:
+since `libcudf` objects are not Python objects, we need something to manage their lifetimes from within cuDF.
+
+The key is to recognize that `libcudf` offers both owning (e.g. `column`) and non-owning (e.g. `column_view`) objects.
+All `libcudf` algorithms accept views as parameters while returning (new) owning objects.
+When calling `libcudf` APIs, cuDF Python constructs views from cuDF `Buffers`.
+When owning objects are returned, cuDF has an rmm object take ownership of that memory and stores that in a `Buffer`.
+The result is that all memory allocated by `libcudf` inside cuDF eventually has ownership transferred to a `Buffer`.
+```
```

Comment:
I think this note is a bit misplaced, where it might be easier to read once the reader has an understanding of the Cython and C++ layers.